apps/api/test/06-report-6.e2e-spec.ts

```diff
@@ -0,0 +1,141 @@
+import { IENApplicantCreateUpdateDTO } from '@ien/common';
+import { INestApplication } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import request from 'supertest';
+
+import { AppModule } from 'src/app.module';
+import { AuthGuard } from 'src/auth/auth.guard';
+import { URLS } from './constants';
+import { canActivate } from './override-guard';
+import {
+  getApplicant,
+  getJob,
+  getMilestone,
+  HEALTH_AUTHORITIES,
+  RECRUITMENT_STAGE_STATUSES,
+} from './report-util';
+
+describe('Report 6 - Registrants in Recruitment Stage', () => {
+  let app: INestApplication;
+  let jobTempId = '';
+  let applicantStatusId = 'NA';
+  let applicantId: string;
+
+  const PHSA = 'Provincial Health Services';
+  const NHA = 'Northern Health';
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    })
+      .overrideGuard(AuthGuard)
+      .useValue({ canActivate })
+      .compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  const getReport6 = async () => {
+    const { body } = await request(app.getHttpServer()).get(URLS.REPORT6);
+    return body;
+  };
+
+  const addApplicant = async (applicant: IENApplicantCreateUpdateDTO) => {
+    const { body } = await request(app.getHttpServer()).post('/ien').send(applicant);
+    return body;
+  };
+
+  // add a job
+  const addJob = async (id: string, ha_pcn: string, job_id: string) => {
+    const addJobUrl = `/ien/${id}/job`;
+    const job = getJob({ ha_pcn, job_id });
+    const { body } = await request(app.getHttpServer())
+      .post(addJobUrl)
+      .expect(({ body }) => (jobTempId = body.id))
+      .send(job);
+
+    return body;
+  };
+
+  // add a milestone
+  const addMilestone = async (id: string, status_id?: string, start_date?: string) => {
+    const addStatusUrl = `/ien/${id}/status`;
+    const milestone = getMilestone({
+      status: status_id || RECRUITMENT_STAGE_STATUSES[1].id,
+      job_id: jobTempId || 'WillHaveJobId',
+      start_date: start_date,
+    });
+
+    const { body } = await request(app.getHttpServer()).post(addStatusUrl).send(milestone);
+    applicantId = id;
+    applicantStatusId = body.id;
+  };
+
+  it('Add Recruitment related statuses for 4 HAs', async () => {
```

Comment:
is there any reason for the test to be run for 4 HAs?

Reply:
Just figured 4 or so would be good instead of running the test for all 9 or 10 HA's